atom/browser/api/atom_api_web_contents.cc

```diff
@@ -215,6 +215,34 @@ void WebContents::HandleKeyboardEvent(
       web_contents(), event);
 }
 
+void WebContents::EnterFullscreenModeForTab(content::WebContents* source,
+                                            const GURL& origin) {
+  auto window = GetWindowFromGuest(source);
+  if (window) {
+    window->SetHtmlApiFullscreen(true);
+    source->GetRenderViewHost()->WasResized();
+    Emit("enter-html-full-screen");
```

Comment:
I think `BrowserWindow` should also emit `enter-html-full-screen` when its webview enters fullscreen?